src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonParameterInfo.cs

```diff
@@ -40,54 +36,42 @@ public JsonClassInfo RuntimeClassInfo
             {
                 if (_runtimeClassInfo == null)
                 {
-                    _runtimeClassInfo = Options.GetOrAddClass(_runtimePropertyType);
+                    _runtimeClassInfo = Options.GetOrAddClass(RuntimePropertyType);
                 }
 
                 return _runtimeClassInfo;
             }
         }
 
+        internal Type RuntimePropertyType { get; set; } = null!;
+
         public bool ShouldDeserialize { get; private set; }
 
         public virtual void Initialize(
-            Type declaredPropertyType,
             Type runtimePropertyType,
             ParameterInfo parameterInfo,
             JsonPropertyInfo matchingProperty,
             JsonSerializerOptions options)
         {
-            _runtimePropertyType = runtimePropertyType;
-
-            Options = options;
-            ParameterInfo = parameterInfo;
+            RuntimePropertyType = runtimePropertyType;
             Position = parameterInfo.Position;
-            ShouldDeserialize = true;
-
-            DetermineParameterName(matchingProperty);
-        }
-
-        private void DetermineParameterName(JsonPropertyInfo matchingProperty)
-        {
             NameAsUtf8Bytes = matchingProperty.NameAsUtf8Bytes!;
+            Options = options;
+            ShouldDeserialize = true;
         }
 
         // Create a parameter that is ignored at run-time. It uses the same type (typeof(sbyte)) to help
         // prevent issues with unsupported types and helps ensure we don't accidently (de)serialize it.
         public static JsonParameterInfo CreateIgnoredParameterPlaceholder(
-            ParameterInfo parameterInfo,
             JsonPropertyInfo matchingProperty,
             JsonSerializerOptions options)
         {
-            JsonParameterInfo jsonParameterInfo = new JsonParameterInfo<sbyte>();
-            jsonParameterInfo.Options = options;
-            jsonParameterInfo.ParameterInfo = parameterInfo;
-            jsonParameterInfo.ShouldDeserialize = false;
-
-            jsonParameterInfo.DetermineParameterName(matchingProperty);
-
-            return jsonParameterInfo;
+            return new JsonParameterInfo<sbyte>
+            {
+                RuntimePropertyType = typeof(sbyte),
+                NameAsUtf8Bytes = matchingProperty.NameAsUtf8Bytes!,
+                Options = options
```

Comment:
The options aren't used. If we remove this line, we might need to add debug asserts where `Options` is used.

Reply:
They are used later when we call TryReadAsObject\TryRead

Reply:
Not when `ShouldDeserialize` is false..

Reply:
I see. Yes will not assign for that case and make Options nullable and add Asserts.